src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { FC } from 'react';
+import styled from 'styled-components';
+import { Spaces } from '../..';
+import { Box, BoxProps } from '../Box/Box';
+
+const COLUMN_WIDTH = 100 / 12;
+const GAP = Spaces[3];
+
+const Row = styled(Box).attrs({ display: 'flex', flexDirection: 'row', flexWrap: 'wrap' })`
+    gap: ${GAP};
+    margin-right: -${GAP};
+`;
+
+type ColumnOffset = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+type ColumnSpan = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+
+interface ColumnProps extends BoxProps {
+    span?: ColumnSpan;
+    offset?: ColumnOffset;
+}
+
+const Column: FC<ColumnProps> = ({ span = 1, offset = 0, ...restProps }: ColumnProps) => {
+    const marginLeft = `${offset * COLUMN_WIDTH}%`;
```

Comment:
Have you considered using a [CSS grid](https://css-tricks.com/snippets/css/complete-guide-grid/) instead of a flexbox?

In my mind, we are trying to make CSS grid out of a flexbox container. All those things are supported in the `display:grid` nicely and we can be just a component that abstracts CSS props.

Reply:
I tried this in a previous iteration (244b393). What I did not like about it is that, to me, it did not made sense to obscure the grid api behind row and column components. I swathes the implementation to flex, as it scopes better that the contents are to be arranged in only one direction.

I really prefer the implementation suing grid, but then I would rework it in a way that we have `Grid` instead of `Row` and `GridItem` instead of `Column`

Reply:
I don't have a strong opinion about it as long as we can use it to create a layout easily, even though it's true that it sounds weird to have a Grid system using flex instead of grid